### ISSUE TITLE
Forms and Record Sets don't switch from/to read-only mode even if gear icon does

### ISSUE THREAD

**Issue by realVinayak** [specify/specify7#2767]
**Describe the bug**
It seems like the gear icon in the Record Sets and Forms listen to the changes to read-only mode from other tabs, however, the change in the other tab doesn't update the page itself. Thus, the gear icon says it is in the read-only mode (based on another tab) but the page isn't in the read-only mode.

The issue is in 7.8.2 and edge

**To Reproduce**
1. Open a form and open another (or same) record set or form in another tab
2. Enable read-only mode in the first form.
3. The gear icon in another tab says it is in read-only mode but the page isn't.

**Expected behavior**
The form/record set in the second tab should have been updated. Also, the save button in the form/record set is still visible even if the component is in the read-only mode (but doesn't do anything) and the user is still able to make changes to the component even if the read-only mode is enabled.

Further, if the user makes a lot of changes to the form but doesn't save them and goes into read-only mode from another tab, then since the first form didn't update the layout (the user still sees editable fields and the save button) they lose those changes. 

An easy fix (but harsh on user) would be to refresh the page whenever the form / record set goes into read-only mode. An easier-on-the-user fix would be to check if the user made any changes to the form before going to the read-only mode (even if they went into read-only mode from another tab)

**Screenshots**
I went into read-only mode from another tab in both of the screenshots.
Editable fields are still visible even if I'm in the read-only mode
![editable field still visible if the user is in readonly mode](https://user-images.githubusercontent.com/61122018/211361226-927d5f94-4e9d-41f7-88c1-28da7266d201.PNG)
Readonly fields are still visible even if I'm not in the read-only mode
![readonly fields visible even if the user isn't in readonly fields](https://user-images.githubusercontent.com/61122018/211361361-eba797f1-ef43-472b-8e59-cf9c9c3e68db.PNG)

**Test Panel Link**
https://purherb-v782.test.specifysystems.org/
https://purherb-edge.test.specifysystems.org/

**Comment by maxxxxxdlp** [comment]
Not sure how likely this to occur, but here is why it happens:
As soon as you change the read-only mode, there is logic in there to reload the page.
Need to instead move that logic into a separate place, and have that place be called by both direct and indirect changes to prefs.

Same bug would probably happen for UI language change and schema language change

**Comment by realVinayak** [comment]
Yep, the UI language change and schema language change are only reflected when the user reloads (if they changed the language from another tab). Each change that causes a reload (read-only mode will be one of them) will have a URL to redirect to and we just call it in prefs and read-only mode.

**Comment by emenslin** [comment]
The main problem of this is fixed however when you have the 2 tabs open and change one to read only it will remove 'data entry' from the navigation bar but if you switch to the other tab 'data entry' will still be on the navigation bar.  And then if you disable read only from the second tab 'data entry' will still be gone from the first tab. Not sure if it should be it's own issue but I figured I would just include it here 

https://github.com/specify/specify7/assets/150940840/f9574315-6cc6-4759-b2a4-2576ba61aad5

**Comment by emenslin** [comment]
Can recreate my previous comment in edge (7.9.6)
